packages/core/src/indexing/insertion-order.ts

```diff
@@ -177,11 +197,45 @@ export class InsertionOrder {
         base = base.where(converted.where)
       }
 
-      if (isReverseOrder) {
-        base = base.orderBy(reverseOrder(INSERTION_ORDER))
+      // Handle cursor if present
```

Comment:
this function is getting very long and hard to read.  Can we break it up into some helpers to improve readability?